stack.py

```diff
@@ -309,10 +316,22 @@ def eval_program(program):
     'gt': lambda a, b: float(b > a),
     'ge': lambda a, b: float(b >= a),
     'lt': lambda a, b: float(b < a),
-    'le': lambda a, b: float(b <= a)
+    'le': lambda a, b: float(b <= a),
+    'and': lambda a, b: 1 if (a and b) else 0,
+    'or': lambda a, b: 1 if (a or b) else 0,
+    'xor': lambda a, b: 1 if ((not a) and b) or (a and (not b)) else 0,
```

Comment:
This might look better as `float(bool(a) != bool(b))`, since `!=` is logical xor for bools.